### PR TITLE
start a node: add clarification on specifying flags every time

### LINKED ISSUE #1813: Clarify --join flag
From @bdarnell:  

`--join` is only *required* when there is no data directory

as long as the data directory is preserved, the previous join targets can be recovered from there

however, it is highly *recommended* to always set `--join`, to avoid mistakes if the data directory has been wiped or if node addresses have changed

cc: @robert-s-lee

### LINKED ISSUE #1855: Start a Node Doc Update
Re: [Start a Node](https://cockroachlabs.com/docs//v1.0/start-a-node.html)

## Issue Description

I had an issue recently where I restarted a node without advertise host, and it was able to join the cluster, but none of the other nodes could talk to it an queries could not run.

## Suggested Resolution 

For restarting a node in a cluster in the cloud, we should make it clear that users need to advertise host.

### PR DESCRIPTION
**Pull request by arjunravinarayan**
Fixes #1855, fixes #1813.

### CODE REVIEW

File: v1.0/start-a-node.md [hunk 1]
```diff
@@ -25,7 +25,13 @@ $ cockroach start --help
 
 ## Flags
 
-The `start` command supports the following [general-use](#general) and [logging](#logging) flags.
+The `start` command supports the following [general-use](#general) and
+[logging](#logging) flags. All flags must be specified each time the
+node is started, as they will not be remembered, with the exception of
+the <code>--join</code> flag. Nevertheless, we recommend specifying
```
Comment (jseldess) [inline v1.0/start-a-node.md]: nit: can you use backticks instead of `<code></code>`?

File: v1.1/start-a-node.md [hunk 2]
```diff
@@ -29,7 +29,13 @@ $ cockroach start --help
 
 ## Flags <span class="version-tag">Changed in v1.1</span>
 
-The `start` command supports the following [general-use](#general) and [logging](#logging) flags.
+The `start` command supports the following [general-use](#general) and
+[logging](#logging) flags. All flags must be specified each time the
+node is started, as they will not be remembered, with the exception of
+the <code>--join</code> flag. Nevertheless, we recommend specifying
```
Comment (jseldess) [inline v1.1/start-a-node.md]: Same as above.

File: v2.0/start-a-node.md [hunk 3]
```diff
@@ -29,7 +29,13 @@ $ cockroach start --help
 
 ## Flags <span class="version-tag">Changed in v1.1</span>
 
-The `start` command supports the following [general-use](#general) and [logging](#logging) flags.
+The `start` command supports the following [general-use](#general) and
+[logging](#logging) flags. All flags must be specified each time the
+node is started, as they will not be remembered, with the exception of
+the <code>--join</code> flag. Nevertheless, we recommend specifying
```
Comment (jseldess) [inline v2.0/start-a-node.md]: Same as above.

**Comment by arjunravinarayan** [comment]
Done.

**Comment by jseldess** [comment]
@arjunravinarayan, you need to update your copy of master, rebase this branch against it, and push again to get past the TC failure. Then we can merge!

**Pull request merged**